feature/bgp/otg_tests/bgp_override_as_path_split_horizon_test/bgp_override_as_path_split_horizon_test.go

```diff
@@ -0,0 +1,581 @@
+	nbr2 = &bgpNeighbor{localAS: dutLocalAS2, peerAS: ateAS2, neighborip: atePort2.IPv4, isV4: true, peerGrp: peerGrpName2}
+
+	otgPort1V4Peer = "atePort1.BGP4.peer"
+	otgPort2V4Peer = "atePort2.BGP4.peer"
+)
+
+// configureDUT configures all the interfaces on the DUT.
+func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
+	t.Helper()
+	dc := gnmi.OC()
+	i1 := dutPort1.NewOCInterface(dut.Port(t, "port1").Name(), dut)
+	gnmi.Replace(t, dut, dc.Interface(i1.GetName()).Config(), i1)
+
+	i2 := dutPort2.NewOCInterface(dut.Port(t, "port2").Name(), dut)
+	gnmi.Replace(t, dut, dc.Interface(i2.GetName()).Config(), i2)
+}
+
+// verifyPortsUp asserts that each port on the device is operating.
+func verifyPortsUp(t *testing.T, dev *ondatra.Device) {
+	t.Helper()
+	for _, p := range dev.Ports() {
+		status := gnmi.Get(t, dev, gnmi.OC().Interface(p.Name()).OperStatus().State())
+		if want := oc.Interface_OperStatus_UP; status != want {
+			t.Errorf("%s Status: got %v, want %v", p, status, want)
+		}
+	}
+}
+
+// bgpCreateNbr creates a BGP object with neighbors pointing to atePort1 and atePort2
+func bgpCreateNbr(t *testing.T, dut *ondatra.DUTDevice) *oc.NetworkInstance_Protocol {
+	t.Helper()
+	dutOcRoot := &oc.Root{}
+	ni1 := dutOcRoot.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut))
+	niProto := ni1.GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP")
+	bgp := niProto.GetOrCreateBgp()
+	global := bgp.GetOrCreateGlobal()
+	global.RouterId = ygot.String(dutPort2.IPv4)
+	global.As = ygot.Uint32(dutGlobalAS)
+
+	// Note: we have to define the peer group even if we aren't setting any policy because it's
+	// invalid OC for the neighbor to be part of a peer group that doesn't exist.
+
+	for _, nbr := range []*bgpNeighbor{nbr1, nbr2} {
+
+		pg := bgp.GetOrCreatePeerGroup(nbr.peerGrp)
+		pg.PeerAs = ygot.Uint32(nbr.peerAS)
+		pg.LocalAs = ygot.Uint32(nbr.localAS)
+		pg.PeerGroupName = ygot.String(nbr.peerGrp)
+
+		nv4 := bgp.GetOrCreateNeighbor(nbr.neighborip)
+		nv4.PeerGroup = ygot.String(nbr.peerGrp)
+		nv4.PeerAs = ygot.Uint32(nbr.peerAS)
+		nv4.Enabled = ygot.Bool(true)
+		af4 := nv4.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST)
+		af4.Enabled = ygot.Bool(true)
+
+		if deviations.RoutePolicyUnderAFIUnsupported(dut) {
+			rpl := pg.GetOrCreateApplyPolicy()
+			rpl.ImportPolicy = []string{policyName}
+			rpl.ExportPolicy = []string{policyName}
+		} else {
+			pgaf := pg.GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST)
+			pgaf.Enabled = ygot.Bool(true)
+			rpl := pgaf.GetOrCreateApplyPolicy()
+			rpl.ImportPolicy = []string{policyName}
+			rpl.ExportPolicy = []string{policyName}
+		}
+	}
+	return niProto
+}
+
+// verifyBgpTelemetry checks that the dut has an established BGP session with reasonable settings.
+func verifyBgpTelemetry(t *testing.T, dut *ondatra.DUTDevice, nbrsList []*bgpNeighbor) {
```

Comment:
Please use cfgplugins.VerifyDUTBGPEstablished https://github.com/openconfig/featureprofiles/blob/520108e9704e82444726ff35d0687f09163f8994/internal/cfgplugins/bgp.go#L316

Reply:
updated this one.